src/nn.h

```diff
@@ -387,9 +401,35 @@ NN_EXPORT int nn_device (int s1, int s2);
 
 NN_EXPORT int nn_tcpmuxd (int port);
 
+/******************************************************************************/
+/*  Statistics.                                                               */
+/******************************************************************************/
+
+/*  Transport statistics  */
+#define NN_STAT_ESTABLISHED_CONNECTIONS 101
+#define NN_STAT_ACCEPTED_CONNECTIONS    102
+#define NN_STAT_DROPPED_CONNECTIONS     103
+#define NN_STAT_BROKEN_CONNECTIONS      104
+#define NN_STAT_CONNECT_ERRORS          105
+#define NN_STAT_BIND_ERRORS             106
+#define NN_STAT_ACCEPT_ERRORS           107
+
+#define NN_STAT_CURRENT_CONNECTIONS     201
+#define NN_STAT_INPROGRESS_CONNECTIONS  202
+#define NN_STAT_CURRENT_EP_ERRORS       203
+
+/*  The socket-internal statistics  */
+#define NN_STAT_MESSAGES_SENT           301
+#define NN_STAT_MESSAGES_RECEIVED       302
+#define NN_STAT_BYTES_SENT              303
+#define NN_STAT_BYTES_RECEIVED          304
+/*  Protocol statistics  */
+#define	NN_STAT_CURRENT_SND_PRIORITY    401
+
+NN_EXPORT NN_UINT64 nn_get_statistic (int s, int stat);
```

Comment:
Turns out this is possible (I had already thought about it). Using the nn_symbol() capability, you can iterate over all symbols, and just look for the ones that are in the NN_NS_STATISTIC namespace.  So everything is discoverable.

Offering up all of them would have been a bit trickier to do in an interface that is meant to evolve (as new stats get added, and possibly old ones deprecated).  This way we can evolve without breaking binaries due to incorrectly compiled structure sizes or somesuch.

And the whole thing is discoverable without me having to invent a new API, which I love.  (Not that I really have any love of the nn_symbol() API -- but in this case it solves the problem for us.)